ddtrace/tracer/civisibility_payload_test.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package tracer
+
+import (
+	"bytes"
+	"io"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tinylib/msgp/msgp"
+)
+
+func newCiVisibilityEventsList(n int) []*ciVisibilityEvent {
+	itoa := map[int]string{0: "0", 1: "1", 2: "2", 3: "3", 4: "4", 5: "5"}
+	list := make([]*ciVisibilityEvent, n)
+	for i := 0; i < n; i++ {
+		s := newBasicSpan("span.list." + itoa[i%5+1])
+		s.Start = fixedTime
+		list[i] = getCiVisibilityEvent(s)
+	}
+
+	return list
+}
+
+// TestCiVisibilityPayloadIntegrity tests that whatever we push into the payload
+// allows us to read the same content as would have been encoded by
+// the codec.
+func TestCiVisibilityPayloadIntegrity(t *testing.T) {
+	want := new(bytes.Buffer)
+	for _, n := range []int{10, 1 << 10, 1 << 17} {
```

Comment:
Personal question, where did the `10, 1 << 10, 1 << 17` numbers come from?

Reply:
I think it was copied from the tracer payload test: https://github.com/DataDog/dd-trace-go/blob/e3400ddee3126d26574ea71eccb47f2d649400d4/ddtrace/tracer/payload_test.go#L37

I'm not sure why those numbers were chosen 😄

Reply:
that's fine 😁